docs/src/pages/getting-started/page-layout-examples/sign-in-side/SignInSide.js

```diff
@@ -61,7 +61,7 @@ export default function SignInSide() {
     <Grid container component="main" className={classes.root}>
       <CssBaseline />
       <Grid item xs={false} sm={4} md={7} className={classes.image} />
-      <Grid item xs={12} sm={8} md={5} component={() => <Paper square elevation={6} />}>
+      <Grid item xs={12} sm={8} md={5} component={Paper} elevation={6} square>
```

Comment:
Just a random thought but would it maybe make more sense to push towards something like

```jsx
function GridPaper() {
  // but something specific here e.g. const className = useGrid(); or const css = useGrid()
  const props = useGrid({ item, xs: 12 });

  return <Paper {...props} />
}```

The current approach tends to grow quite nasty (member hocs?) with regards to what props belongs to what e.g. `<Grid item elevation={12] component={Paper} />` (is elevation a Paper or Grid prop?).

IMO `components` should only ever accept strings for host components. There are a few exceptions like Links but I'd hope these will be replace with hooks in the future.

Reply:
Some thought:
- useGrid would require a hook ready CSS-in-JS solution.
- This example don't need to use the component props. We could refactor it.
- This example is lazy, it could create an intermediary component.

Reply:
Yeah I'm not really sure about the best API. IMO this `<Render component={Something} />` pattern is just disguised hoc. I might've been a good idea for router links but I think it got abused<sup>1</sup> beyond that.

<sup>1</sup> At least now we might be able to provide a better API with hooks.